api/src/main/java/com/community/api/endpoint/avisoft/controller/cart/CartEndPoint.java

```diff
@@ -126,21 +127,34 @@ public ResponseEntity<?> retrieveCartItemsCount(@RequestParam long customerId, @
             return new ResponseEntity<>("Error retrieving cart", HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+    @Transactional
+    @JsonBackReference
```

Comment:
also

from now wards
try to use
autowiring like using constructor based

    @Autowired
    public CartEndPoint(CustomerService customerService, OrderService orderService, CatalogService catalogService,
                        ExceptionHandlingImplement exceptionHandling, EntityManager entityManager,
                        OrderItemService orderItemService, CartService cartService) {
        this.customerService = customerService;
        this.orderService = orderService;
        this.catalogService = catalogService;
        this.exceptionHandling = exceptionHandling;
        this.entityManager = entityManager;
        this.orderItemService = orderItemService;
        this.cartService = cartService;
    }